src/platform/Infineon/CYW30739/KeyValueStoreManagerImpl.cpp

```diff
@@ -176,13 +176,12 @@ CHIP_ERROR KeyValueStoreManagerImpl::EraseAll(void)
 
 KeyValueStoreManagerImpl::KeyStorage::KeyStorage(const char * key) : mValueSize(0)
 {
+    memset(mKey, 0, sizeof(mKey));
+
     if (key != NULL)
     {
-        Platform::CopyString(mKey, key);
-    }
-    else
-    {
-        mKey[0] = 0;
+        /* Null-terminated key isn't needed by strncmp in IsMatchKey */
+        strncpy(mKey, key, sizeof(mKey));
```

Comment:
When reading code, I would generally assume anything `char[N]` to be null-terminated string.

If this is not the case, how about changing mKey to be `uint8_t mKey[PersistentStorageDelegate::kKeyLengthMax]`.

This WILL have significant ripple effects, but that shows how confusing a non-null-terminated char array is (when all the values passed into KVS seem to be null terminated.

Alternatively make this string null terminated and allocate `PersistentStorageDelegate::kKeyLengthMax + 1`

Reply:
Okay, I made mKey string null-terminated by adding 1 more byte to the storage.